tests/unit/example.test.js

```diff
@@ -0,0 +1,34 @@
+// Suite
+// -----------------------------------------------------------------------------
+describe(`Example Tests`, function() {
+  // Tests
+  // ---------------------------------------------------------------------------
+  test('javascript (node)', async () => {
+    function add(...addends) {
+      return addends.reduce(
+        (accumulator, currentValue) => accumulator + currentValue
+      );
+    }
+    const testValue = add(1, 2, 3);
+
+    expect(testValue).toBe(6);
+  });
+
+  test('dom manipulation (jsdom)', async () => {
```

Comment:
If we can run unit tests in a browser, we can remove jsdom.

Reply:
JSDom is bundled with Jest, so we can't remove it. It's there if you want to use it, but you don't have to.

Reply:
> @jhildenbiddle yep, I see, I just make sure the complete folder u mentioned.
> per to the test script tests, it works well. cool !
> 
> A small suggestion, the default `8080` port is conflict with another apps server easily(I met it 😅 ).
> If there would change to another port like `8888` for avoiding unnacessary exception at e2e tests.

Makes sense I'll make the change.

Thoughts on `8181`, `8888`, `3000`, or `3001`?

Reply:
> Thoughts on 8181, 8888, 3000, or 3001?

IMO,  any one of them is ok,except 3000.
cuz docsify default port is 3000.

Reply:
> FWIW, I'd prefer if folks focused on e2e tests in general, since we get the most bang-for-our-buck there. We can test 50 function and still not guarantee that a site renders properly. For example, a single e2e test can answer the question "is the sidebar doing X when setting Y is true` much faster and easier than multiple unit or integration tests.

We need both I think. There's reason for each (which we should all learn about so we test things the right way).

For example, in #1227 there is [this test](https://github.com/docsifyjs/docsify/pull/1227/files#diff-b26412a9e2fa48d069811ef18906adf3R8) that tests that `isExternal` does what it actually should do. I specifically put test cases there that I believe the current `isExternal` will fail at (as well some others). Unit tests are good there, and it is probably something that we're unlikely to catch with e2e tests.

Reply:
@trusktr --

Agreed. Those tests are unit tests, not e2e tests. The follow-up to that statement is that unit tests won't determine if `isExternal()` will fail in Firefox or Webkit because those unit tests are tested in Node (which, along with Chrome, use the V8 JavaScript engine). An e2e test may not cover all scenarios like unit tests do, but testing `isExternal()` behavior with an e2e test using one external and one internal URL is a good idea as well.

Side note: What is the purpose of the `initJSDOM()` function in those tests? Maybe I'm missing something, but I believe those `expect` assertions are being run in Node and aren't associated with jsdom.